### ISSUE TITLE
/mob/new_player doesn't qdel.

### ISSUE THREAD

**Issue by MrStonedOne** [tgstation/tgstation#11365]
This happens if there is still a variable that points to that Object. in order to soft delete properly, the Object's destroy proc needs to clear up all references that point to it.

Failed qdel log:
http://privatepaste.com/edc18b668d

**Comment by Fox-McCloud** [comment]
This really makes me wonder if all moves should just be hard deleted instead of all living mobs.

**Comment by lzimann** [comment]
[19:11:41] <Cyberboss> LeoZ #11365 mobs always harddel, that's closable

**Comment by MrStonedOne** [comment]
Nope. This happened because the mob tried to soft delete and failed

**Comment by lzimann** [comment]
@Cyberboss

**Comment by Cyberboss** [comment]
@MrStonedOne At some point mobs were considered too complex to attempt GCing so some one changed the hint for all mobs to QDEL_HINT_HARDDEL https://github.com/tgstation/tgstation/blob/master/code/modules/mob/mob.dm#L20

**Comment by MrStonedOne** [comment]
Ya, that happened BEFORE this issue report

mobs have *always* hard deleted, since qdel was even thing.

so, tell me:

WHY ARE NEW PLAYERS TRYING TO SOFT DELETE?

**Comment by Cyberboss** [comment]
Best guess is mob/Destroy crashed somewhere, because there is no /mob/dead/new_player/Destroy now

**Comment by MrStonedOne** [comment]
Its an old issue report, so it could very well be fixed by now, but not for the reasons you claimed (since what you claimed had always been the case) so i had it re-opened.

**Comment by vuonojenmustaturska** [comment]
Appending my duplicate bug report here, for extra data:

using QDEL_HINT_FINDREFERENCE, the remaining references are:
`\## TESTING: Found /mob/dead/new_player [0x3000105] in list World -> Garbage -> list -> list.
\## TESTING: Found /mob/dead/new_player [0x3000105] in /client's movingmob var. World -> Naksuasdf`

The second one seems to be the important one. I don't have enough experience with the client-swapping bits of byond/tg to start untangling it without introducing shittons of bugs that might not be evident in local testing.

[Directions]: 
define TESTING and GC_FAILURE_HARD_LOOKUP, start a new game and join as any job, eventually the /mob/dead/new_player will fail to del. Or use QDEL_HINT_FINDREFERENCE.

**Comment by MrStonedOne** [comment]
I can't find where movingmob is defined and its not listed in the byond ref.

**Comment by Cyberboss** [comment]
It's in parallax.dm

**Comment by nightred** [comment]
This issue is so large and nebulas that it will never be closed.
Please reopen with a more exact failing to qdel error or location.
No one has made an effor or touch this for years because of the scope and the statement that it is just failing.

**Comment by Qustinnus** [comment]
literaly a pr open for this.

also closing an issue mso made while its not fixed yet?  damn wish i was this ballsy

**Comment by Jared-Fogle** [comment]
today on sybil

![image](https://user-images.githubusercontent.com/35135081/100494527-71d87a80-30f7-11eb-9810-5a8378c15a60.png)

**Comment by LemonInTheDark** [comment]
```
## TESTING: Beginning search for references to a /mob/dead/new_player.
## TESTING: Found /mob/dead/new_player [0x3000009] in /datum/mind's current var. World -> /datum/mind
## TESTING: Completed search for references to a /mob/dead/new_player.
## TESTING: GC: -- (VV) | /mob/dead/new_player was unable to be GC'd --
## TESTING: GC: -- [0x21000004] | /mob/dead/new_player was unable to be GC'd --
```
who wants to clear mind.current automatically

**Comment by LemonInTheDark** [comment]
I was wrong to be hopeful
![image](https://user-images.githubusercontent.com/58055496/100700085-7dbf7900-3351-11eb-85d2-d1d14c8bc98a.png)

**Comment by Jared-Fogle** [comment]
am i crazy to think mind.current should be a weakref anyway?

**Comment by Jared-Fogle** [comment]
![image](https://user-images.githubusercontent.com/35135081/101227135-b6aa6700-364b-11eb-8531-f2269838c9d2.png)

we're so close

**Comment by LemonInTheDark** [comment]
depression

**Comment by Azarak** [comment]
![obraz](https://user-images.githubusercontent.com/6381979/102506263-40622780-4083-11eb-937e-73f7f2716e56.png)
fix does work

**Comment by Jared-Fogle** [comment]
3rd times the charm

**Comment by Jared-Fogle** [comment]
![image](https://user-images.githubusercontent.com/35135081/102704884-08e7ac80-4236-11eb-921d-357a029e2c24.png)


The visions dancing in my mind 
The early dawn, the shades of time 
Twilight crawling through my windowpane

**Comment by Azarak** [comment]
Oh noooooo

**Comment by LemonInTheDark** [comment]
![hhh](https://user-images.githubusercontent.com/58055496/128620418-9b361b12-777a-46d8-980f-ebbe72daee61.png)
WHY WON'T IT DIE
For a more upclose view of that last little bit
![hhh](https://user-images.githubusercontent.com/58055496/128620478-5c754184-c21e-41e5-8c8f-08f276dd5673.png)

**Comment by Terra1** [comment]
Confirmed still existing as of round ID 262018

**Comment by Frfor17** [comment]
11 Years?

**Comment by LemonInTheDark** [comment]
11 years
